apps/api/src/TransformUser/TransformUser.service.ts

```diff
@@ -0,0 +1,75 @@
+import { User, Userfront } from 'types';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Injectable, BadRequestException } from '@nestjs/common';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class TransformUserService {
+  constructor(
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+  ) {}
+
+  async #getVictoryNumber(user_id: string): Promise<number> {
+    const user = await this.userRepository.findOne({
+      relations: {
+        win: true,
+      },
+      where: {
+        id: user_id,
+      },
+    });
+    if (!user) throw new BadRequestException('user not found');
+
+    return user.win.length;
+  }
+
+  async #getDefeatNumber(user_id: string): Promise<number> {
+    const user = await this.userRepository.findOne({
+      relations: {
+        defeat: true,
+      },
+      where: {
+        id: user_id,
+      },
+    });
+    if (!user) throw new BadRequestException('user not found');
+
+    return user.defeat.length;
+  }
+
+  async #getRank(user_id: string): Promise<number> {
```

Comment:
I think you can simplify things with count and in your query add a where level LessThan:
`return await this.userRepository.count({
where:
{
 level: LessThan(currentUser.level);
}})`

Reply:
done